components/brave_wallet/browser/simulation_service.cc

```diff
@@ -0,0 +1,311 @@
+    return "polygon/v0/mainnet/scan";
+  } else if (coin == mojom::CoinType::ETH &&
+             chain_id == mojom::kBinanceSmartChainMainnetChainId) {
+    return "bnb/v0/mainnet/scan";
+  } else if (coin == mojom::CoinType::ETH &&
+             chain_id == mojom::kArbitrumMainnetChainId) {
+    return "arbitrum/v0/one/scan";
+  }
+
+  return "";
+}
+
+bool HasTransactionScanSupportInternal(const std::string& chain_id,
+                                       mojom::CoinType coin) {
+  return (coin == mojom::CoinType::SOL && chain_id == mojom::kSolanaMainnet) ||
+         (coin == mojom::CoinType::SOL && chain_id == mojom::kSolanaTestnet) ||
+         (coin == mojom::CoinType::SOL && chain_id == mojom::kSolanaDevnet) ||
+         (coin == mojom::CoinType::ETH && chain_id == mojom::kMainnetChainId) ||
+         (coin == mojom::CoinType::ETH && chain_id == mojom::kGoerliChainId) ||
+         (coin == mojom::CoinType::ETH &&
+          chain_id == mojom::kPolygonMainnetChainId) ||
+         (coin == mojom::CoinType::ETH &&
+          chain_id == mojom::kBinanceSmartChainMainnetChainId) ||
+         (coin == mojom::CoinType::ETH &&
+          chain_id == mojom::kArbitrumMainnetChainId);
+}
+
+bool HasMessageScanSupportInternal(const std::string& chain_id,
+                                   mojom::CoinType coin) {
+  return (coin == mojom::CoinType::ETH && chain_id == mojom::kMainnetChainId) ||
+         (coin == mojom::CoinType::ETH && chain_id == mojom::kGoerliChainId) ||
+         (coin == mojom::CoinType::ETH &&
+          chain_id == mojom::kPolygonMainnetChainId) ||
+         (coin == mojom::CoinType::ETH &&
+          chain_id == mojom::kBinanceSmartChainMainnetChainId) ||
+         (coin == mojom::CoinType::ETH &&
+          chain_id == mojom::kArbitrumMainnetChainId);
+}
+
+const base::flat_map<std::string, std::string> GetHeaders() {
+  return {{kBlowfishAPIVersionHeader, kBlowfishAPIVersion},
+          {net::HttpRequestHeaders::kContentType, "application/json"},
+          {kBraveServicesKeyHeader, BUILDFLAG(BRAVE_SERVICES_KEY)}};
+}
+
+}  // namespace
+
+SimulationService::SimulationService(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : api_request_helper_(GetNetworkTrafficAnnotationTag(),
+                          url_loader_factory) {}
+
+SimulationService::~SimulationService() = default;
+
+mojo::PendingRemote<mojom::SimulationService> SimulationService::MakeRemote() {
+  mojo::PendingRemote<mojom::SimulationService> remote;
+  receivers_.Add(this, remote.InitWithNewPipeAndPassReceiver());
+  return remote;
+}
+
+void SimulationService::Bind(
+    mojo::PendingReceiver<mojom::SimulationService> receiver) {
+  receivers_.Add(this, std::move(receiver));
+}
+
+// static
+GURL SimulationService::GetScanTransactionURL(const std::string& chain_id,
+                                              mojom::CoinType coin) {
+  std::string spec = base::StringPrintf(
```

Comment:
Prefer doing it as part of https://github.com/brave/brave-browser/issues/24401, because it looks like we'll need a builder-style helper function like `GURL AppendPath(const GURL& url, base::StringPiece path)` and unit-tests for the same.

By the way, there was a missing `net::AppendQueryParameter` so I added it here: https://github.com/brave/brave-core/pull/18302/commits/81c36ef0cb9e67c0e4bcfa3e3e7849fdcee3fc93

Reply:
This is not security blocking issue because the string text used are controlled constants that don't have the possibility to introduce path traversal issues here.

Related to completing https://github.com/brave/brave-browser/issues/24401, Could we raise the priority on cleaning that up within the wallet then if we plan to introduce new known points where it's being done? Seems a bit off to recognize we need to refactor this here, but then to also continue doing it.